### ISSUE TITLE
In the example, a task is created with status: null instead of open

### ISSUE THREAD

**Issue by SEGVeenstra** [pinchbv/floor#763]
This means the filter in the example is set to 'open', new tasks will not be shown.

When the user swipes to update the status, the status will jump to `inProgress`, thus skipping `open`.

**Comment by dkaera** [comment]
@SEGVeenstra in the example I wanted to show that null values are also supported. Your option makes sense if it looks more visual for general understanding.

**Comment by dkaera** [comment]
@SEGVeenstra 
yes, I agree with you. Adding the `uncategorized` filter should give more understanding.

**Comment by SEGVeenstra** [comment]
@dkaera Ok, I will update the PR probably tomorrow.

**Comment by SEGVeenstra** [comment]
Wasn't able to work on it last week, but I checked it out just now.
Adding the `uncategorized` creates more work than I would like.

Currently the menu items are generated by the possible categories (open, In Progress, Done), with one extra, for showing 'All'.

This is being translated to the selected `TaskStatus`. For this `null` is already used for 'All' and thus cannot be used as filter.

I've made some changes to support the uncategorized option in the PR.

I also noticed the unused property: TaskType.
Maybe we could think about a more excessive example showing a bit more in the future.

**Comment by SEGVeenstra** [comment]
The test for the example fails on the analyzer.
I've fixed all but the last one, about the `library_private_types_in_public_api` rule.
Also mentioned in: #742
